### PR TITLE
Frontend Instrumentation

### PR DESCRIPTION
**Pull request by garimaprachi**

### CODE REVIEW

File: drupal_commerce_razorpay.module [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * Implements hook_page_attachments().
+ */
+function drupal_commerce_razorpay_page_attachments(array &$attachments)
+{
+    if (\Drupal::service('router.admin_context')->isAdminRoute())
+    {
+        $attachments['#attached']['library'][] = 'drupal_commerce_razorpay/admin_styling';
+    }
+}
```
Comment (abdulwahidsharief) [inline drupal_commerce_razorpay.module:15]: @garimaprachi add EOL

File: js/admin-script.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+document.addEventListener("focusout", function (event) {
+    if(event.target.matches("#drupal_razorpay_key_id")) {
+        var data = {
+            'plugin_name': 'drupal',
+			'event' : 'formfield.interacted',
+            'page_url' : window.location.href,
+            'field_type' : 'string',
+            'field_name' : 'drupal_razorpay_key_id'
+		};
```
Comment (abdulwahidsharief) [inline js/admin-script.js:9]: @garimaprachi fix indentation

File: src/Plugin/Commerce/PaymentGateway/RazorpayCheckout.php [hunk 3]
```diff
@@ -112,6 +131,7 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
             '#default_value' => $this->configuration['payment_action'],
         ];
 
+
```
Comment (abdulwahidsharief) [inline src/Plugin/Commerce/PaymentGateway/RazorpayCheckout.php:134]: remove blank lines

File: drupal_commerce_razorpay.module (hunk 1 above)
Comment (garimaprachi) [reply drupal_commerce_razorpay.module:15]: If I add an extra line it starts failing

Comment (garimaprachi) [reply drupal_commerce_razorpay.module]: Done

File: js/admin-script.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        rzpFetch(data);
+    }
+
+    if(event.target.matches("#drupal_razorpay_key_secret")) {
+        var data = {
+            'plugin_name': 'drupal',
+			'event' : 'formfield.interacted',
```
[the remark this answers is not in the document]
Comment (garimaprachi) [reply js/admin-script.js]: Done

File: js/admin-script.js (hunk 2 above)
Comment (garimaprachi) [reply js/admin-script.js]: Done

File: js/admin-script.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+})
+
+function rzpFetch(data) {
+    var body = {
+        mode: 'test',
+        key: "2Ea4C263F7bb3f3AF7630DC5db9e38ff",
+        context: {},
+        events: [
+            {
+                mode: 'test',
```
Comment (abdulwahidsharief) [inline js/admin-script.js:36]: @garimaprachi check mode if it is required here

File: js/admin-script.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+{
+    var data = {
+        'plugin_name': 'drupal',
+		'action' : 'rzpInstrumentation',
+		'event' : 'login.initiated',
+		'next_page_url' : 'https://dashboard.razorpay.com/signin?screen=sign_in&source=drupal'
+	};
+
+    rzpFetch(data);
+}
```
Comment (abdulwahidsharief) [inline js/admin-script.js:74]: @garimaprachi add EOL

File: js/admin-script.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                                                            headers: {
+                                                                },
+        });
+}
+
+function rzpSignupClicked()
+{
+    var data = {
+        'plugin_name': 'drupal',
+		'event' : 'signup.initiated',
```
Comment (abdulwahidsharief) [inline js/admin-script.js:57]: @garimaprachi fix indentation

File: src/Plugin/Commerce/PaymentGateway/RazorpayCheckout.php [hunk 8]
```diff
@@ -83,15 +83,33 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
...
-            <a href="https://dashboard.razorpay.com/signin?screen=sign_in&source=drupal" target="_blank">login</a> if you have an existing account.</p>';
+        $form['signup'] = [
+            '#type' => 'html_tag',
+            '#tag' => 'div',
+            '#attributes' => [
+                'onclick'=>'rzpSignupClicked();'
+            ],
+            '#value' => $this->t('First <a href=":signup_url" target="_blank">Signup</a> for a Razorpay account.',
+                [':signup_url' => 'https://easy.razorpay.com/onboarding?recommended_product=payment_gateway&source=drupal']),
+        ];
```
Comment (abdulwahidsharief) [inline src/Plugin/Commerce/PaymentGateway/RazorpayCheckout.php:94]: @garimaprachi event should only get triggered when user click on link.

File: js/admin-script.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            headers: {
+            },
+        }
+    );
+}
+
+function rzpSignupClicked()
+{
+    var data = {
+        'plugin_name': 'drupal',
```
Comment (abdulwahidsharief) [inline js/admin-script.js:51]: @garimaprachi leave space before :

File: js/admin-script.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        'next_page_url' : 'https://easy.razorpay.com/onboarding/?recommended_product=payment_gateway&source=drupal'
+    };
+
+    rzpFetch(data);
+}
+
+function rzpLoginClicked()
+{
+    var data = {
+        'plugin_name': 'drupal',
```
Comment (abdulwahidsharief) [inline js/admin-script.js:62]: @garimaprachi leave space before :

**Review by abdulwahidsharief — APPROVED** [review]
LGTM

**Pull request merged**